core/src/main/java/com/datastax/oss/driver/internal/core/metadata/MetadataManager.java

```diff
@@ -437,30 +437,35 @@ private void startSchemaRequest(CompletableFuture<RefreshSchemaResult> refreshFu
                     refreshFuture.completeExceptionally(agreementError);
                   } else {
-                    schemaQueriesFactory
-                        .newInstance()
-                        .execute()
-                        .thenApplyAsync(this::parseAndApplySchemaRows, adminExecutor)
-                        .whenComplete(
-                            (newMetadata, metadataError) -> {
-                              if (metadataError != null) {
-                                refreshFuture.completeExceptionally(metadataError);
-                              } else {
-                                refreshFuture.complete(
-                                    new RefreshSchemaResult(newMetadata, schemaInAgreement));
-                              }
-
-                              firstSchemaRefreshFuture.complete(null);
-
-                              currentSchemaRefresh = null;
-                              // If another refresh was enqueued during this one, run it now
-                              if (queuedSchemaRefresh != null) {
-                                CompletableFuture<RefreshSchemaResult> tmp =
-                                    this.queuedSchemaRefresh;
-                                this.queuedSchemaRefresh = null;
-                                startSchemaRequest(tmp);
-                              }
-                            });
+                    try {
+                      schemaQueriesFactory
+                              .newInstance()
+                              .execute()
+                              .thenApplyAsync(this::parseAndApplySchemaRows, adminExecutor)
+                              .whenComplete(
+                                      (newMetadata, metadataError) -> {
+                                        if (metadataError != null) {
+                                          refreshFuture.completeExceptionally(metadataError);
+                                        } else {
+                                          refreshFuture.complete(
+                                                  new RefreshSchemaResult(newMetadata, schemaInAgreement));
+                                        }
+
+                                        firstSchemaRefreshFuture.complete(null);
+
+                                        currentSchemaRefresh = null;
+                                        // If another refresh was enqueued during this one, run it now
+                                        if (queuedSchemaRefresh != null) {
+                                          CompletableFuture<RefreshSchemaResult> tmp =
+                                                  this.queuedSchemaRefresh;
+                                          this.queuedSchemaRefresh = null;
+                                          startSchemaRequest(tmp);
+                                        }
+                                      });
+                    } catch (Throwable t) {
```

Comment:
Was considering RuntimeException but ultimately decided Throwable was better for this use case.
Also, a couple notes here:
- the diff is mostly whitespace since this block is now wrapped with a try/catch
- the only bits that can throw at this level (vs inside the completion handlers) are the `newInstance()` and `execute()` calls, but this felt a lot cleaner than extracting those two bits for the try/catch wrap and then nullchecking the result before the `thenApplyAsync` and `whenComplete`